### ISSUE TITLE
How to calculate accuracy@20 in each  frequency group?

### ISSUE THREAD

**Issue by oldpants** [mp2893/gram#11]
Hi,Choi:
 I have some problems about how to calculate the accuracy@k score of each frequency group. I don't know which of the following two is right: 1.For each frequency group, the top20 score are selected to compare with the real label,and then calculate accuracy@20 individually. 2.Select the in the top 20 index of all labels,and then calculate which group the 20 indexs belong to,and compare with the labels. I hope you can help me if you know.

Many thanks,
Oldpants

**Comment by mp2893** [comment]
Hi Oldpants,

Thanks for taking interest in my work.
It's difficult for me to understand your question, so let me just describe what I did.

Basically, what I did to calculate acc@20 is:
1. Divide the labels into 5 groups according to their frequency in the training data (e.g. 0-20th percentile, 20-40th percentile, ... , 80-100th percentile)
2. Within each group, calculate Accuracy@20 for each label (i.e. disease code). Then calculate the average acc@20 for each group. 

Hope this clears things up. 

Best,
Ed

**Comment by sj2022** [comment]
Hello Ed,

I have a question about how you calculate accuracy@k for this work. Say, we have a set of codes for 80 percentile {A,B,C}.
If the code A is true value and is predicted in top 20, then it is regarded as true positive. If A is not in real value but is predicted, it is false positive and so on. Would you be able to confirm this is the right way or not? 
It will be very appreciated. Thank you in advance.

Best regards,
SJ

**Comment by mp2893** [comment]
Hi SJ, yep that's how I calculated Acc@k.
Best,
Ed

**Comment by sj2022** [comment]
Hi Ed,

Thank you for your prompt reply. In that case, even though I have used diagnose codes in 80-100 percentile group (about total number of 180), there are still a lot of trueNegtaive cases and few truePositive cases, which leads to very higher accuracy than the result from your paper. I have calculated the accuracy for all 180 diagnoses codes and then calculate the average, and receive 88% accuracy.

Best regards,
SJ

**Comment by mp2893** [comment]
By "Dx codes in 80-100 percentile group", do you mean codes that occur frequently in the training set? In that case, I guess it's a bit strange that there are many TNs compared to TPs (since we are talking about frequent codes). For example, if the occurrence probability of code A is 20% in each visit, then on average, the ratio between TNs and TPs cannot be more than 4:1 (though the exact ratio would depend on the k value of Acc@k)

**Comment by sj2022** [comment]
I am sorry that I'm still confused. How about diagnosis codes in very rare
cases (0-20 percentile group). For MIMIC-III, the occurrence of diagnosis
with 80 percentile in training data (the total number of visits = 14941) is
151, which accounts for only 1% (151/14941=0.01). In that case, there will
be a lot of TNs than TPs, which leads to high accuracy and is different
from your result. Sorry for many questions and thank you very much for your
time.


Best regards,
SJ
